File: server/modules/terminal/pty.ts
export function parseResize(msg: string): { cols: number; rows: number } | null {
  const match = msg.match(/^\x1b\[RESIZE:(\d+);(\d+)\]$/);
  if (!match) return null;
  return { cols: parseInt(match[1], 10), rows: parseInt(match[2], 10) };
}

File: server-rs/src/terminal/pty.rs
use regex::Regex;
use std::sync::OnceLock;

pub struct Resize {
    pub cols: u16,
    pub rows: u16,
}

fn resize_re() -> &'static Regex {
    static RE: OnceLock<Regex> = OnceLock::new();
    RE.get_or_init(|| Regex::new(r"^\x1b\[RESIZE:(\d+);(\d+)\]$").expect("resize regex"))
}

pub fn parse_resize(msg: &str) -> Option<Resize> {
    let caps = resize_re().captures(msg)?;
    let cols = caps.get(1)?.as_str().parse().ok()?;
    let rows = caps.get(2)?.as_str().parse().ok()?;
    Some(Resize { cols, rows })
}
